### PR TITLE
Mempool add APIs

### PR DESCRIPTION
**Pull request by al8n**

### CODE REVIEW

File: nodes/nomos-node-api/src/http/backend/axum.rs [hunk 1]
```diff
@@ -256,3 +261,47 @@ where
         Err(e) => (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()).into_response(),
     }
 }
+
+#[utoipa::path(
+    post,
+    path = "/mempool/add/tx",
+    responses(
+        (status = 200, description = "add transaction to the mempool"),
```
Comment (danielSanchezQ) [inline nodes/nomos-node-api/src/http/backend/axum.rs:269]: We need to be consistent in descriptions. I have seen usually we capitalize them. Tiny details.

File: nodes/nomos-node-api/src/http/mempool.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            reply_channel: sender,
+        })
+        .await
+        .map_err(|(e, _)| e)?;
+
+    match receiver.await {
+        Ok(Ok(())) => Ok(()),
+        Ok(Err(())) => Err("mempool error".into()),
+        Err(e) => Err(e.into()),
+    }
```
Comment (zeegomo) [inline nodes/nomos-node-api/src/http/mempool.rs:84]: We don't need two different functions for this, they're the same service with different generics

File: nodes/nomos-node-api/src/http/backend/axum.rs [hunk 3]
```diff
@@ -94,6 +97,8 @@ where
             .route("/carnot/info", routing::get(carnot_info::<T, S, SIZE>))
             .route("/network/info", routing::get(libp2p_info))
             .route("/storage/block", routing::post(block::<S, T>))
+            .route("/mempool/add/tx", routing::post(add_tx::<T>))
+            .route("/mempool/add/cert", routing::post(add_cert))
```
[the remark this answers is not in the document]
Comment (zeegomo) [reply nodes/nomos-node-api/src/http/backend/axum.rs:105]: The concept of a mempool is the same for both, I don't have strong opinions on this but there might be suggestions on the OpenAPI standard on how to handle it

File: nodes/nomos-node-api/src/http/backend/axum.rs [hunk 4]
```diff
@@ -30,6 +31,8 @@ pub struct AxumBackendSettings {
     pub carnot: OverwatchHandle,
     pub network: OverwatchHandle,
     pub storage: OverwatchHandle,
+    pub cert_mempool: OverwatchHandle,
```
Comment (zeegomo) [inline nodes/nomos-node-api/src/http/backend/axum.rs:34]: why do we need to keep adding handles?

**Review by zeegomo — CHANGES_REQUESTED** [review]

**Review by danielSanchezQ — APPROVED** [review]

**Pull request merged**